### PR TITLE
WIP global: recaptcha for invenio-userprofiles

### PR DESCRIPTION
**Pull request by bruno314**
* Adds optional recaptcha during user registration.

Signed-off-by: Bruno Cuc <bruno.cuc@cern.ch>

### CODE REVIEW

**Comment by bruno314** [comment]
In this simplified version rebased onto Sami's commit verification works. Now working on making it independent

File: examples/app.py [hunk 1]
```diff
@@ -94,7 +94,12 @@
     ACCOUNTS_USE_CELERY=False,
     BABEL_DEFAULT_LOCALE='en',
     I18N_TRASNLATION_PATHS=[
-        messages_path(), ]
+        messages_path(), ],
+    # TODO is there public combo of pub/priv. key for HW apps/demos??
+    RECAPTCHA_PUBLIC_KEY="6LcFCBcTAAAAAGCHz2_Pa1AEnY9GJwmix8nd_Z1R",
```
Comment (jirikuncar) [inline examples/app.py]: allow people to configure them via `os.environ` and prepare documentation how to obtain `RECAPTCHA` keys.

File: invenio_userprofiles/ext.py [hunk 2]
```diff
@@ -58,6 +58,7 @@ def init_config(self, app):
         """Initialize configuration."""
         app.config.setdefault('USERPROFILES', True)
         app.config.setdefault('USERPROFILES_EXTEND_SECURITY_FORMS', False)
+        app.config.setdefault('USERPROFILES_EXTEND_RECAPTCHA', False)
```
Comment (jirikuncar) [inline invenio_userprofiles/ext.py]: remove

File: invenio_userprofiles/forms.py [hunk 3]
```diff
@@ -54,6 +55,8 @@ def current_user_email(form, field):
 
 class ProfileForm(Form):
     """Form for editing user profile."""
+    
+    recaptcha = RecaptchaField()
```
Comment (jirikuncar) [inline invenio_userprofiles/forms.py]: add to `register_form_factory` with `if app.config.get('USERPROFILES_PUBLIC_KEY') and ...(...PRIVATE_KEY'):` guard

File: invenio_userprofiles/forms.py [hunk 4]
```diff
@@ -138,4 +141,4 @@ class ConfirmRegisterForm(Form):
 
         profile = FormField(ProfileForm, separator='.')
 
-    return ConfirmRegisterForm
+    return ConfirmRegisterForm
```
Comment (jirikuncar) [inline invenio_userprofiles/forms.py]: configure your editor to terminate file with `\n\0`!

**Comment by lnielsen** [comment]
Supersede by inveniosoftware/invenio-accounts#111

**Pull request closed without merging**